### ISSUE TITLE
ktest reports ERRORs in a wrong way

### ISSUE THREAD

**Issue by grosu** [kframework/k-legacy#518]
One of the specific design aspects of ktest was to report its ERRORs in such a way that one copies the command in the error and pastes it in a terminal to execute the command to see its output.  That was VERY convenient.  Now, ktest outputs errors as follows:

```
fslwork$ ktest tests/config.xml 
Running initial scripts...
SUCCESS
Kompile the language definitions...(2 in total)
SUCCESS
Generate PDF files...(2 in total)
SUCCESS
Running /home/grosu/github.grosu/k/tutorial/1_k/4_imp++/lesson_3/imp.k programs... (8 in total, 0 with input, 8 with output, 0 with error)
ERROR: ['/home/grosu/github.grosu/k/bin/krun' '/home/grosu/github.grosu/k/tutorial/1_k/4_imp++/lesson_1/tests/spawn.imp'] output doesn't match with expected output (time: 2392 ms)
...
ERROR: ['/home/grosu/github.grosu/k/bin/krun' '/home/grosu/github.grosu/k/tutorial/1_k/4_imp++/lesson_1/tests/io.imp' '--directory' 'imp-java_kompiled'] failed with error (time: 163 ms)
...
```

Please give us the previous functionality.  I'm assigning this to Daejun, because it was him who implemented it first place.

**Comment by grosu** [comment]
@daejunpark or @osa1, can you guys please fix this bug asap?  The errors reported by ktest should be of the form 

```
ERROR: [/home/grosu/github.grosu/k/bin/krun /home/grosu/github.grosu/k/tutorial/1_k/4_imp++/lesson_1/tests/spawn.imp] output doesn't match with expected output (time: 2392 ms)
```

and NOT of the form

```
ERROR: ['/home/grosu/github.grosu/k/bin/krun' '/home/grosu/github.grosu/k/tutorial/1_k/4_imp++/lesson_1/tests/spawn.imp'] output doesn't match with expected output (time: 2392 ms)
```

The rationale for that error message was that one can copy the contents in the ERROR's brackets, paste it in console and thus execute the EXACT krun command that ktest executed.  One thing is the additional quotes, another is that the --directory option is missing; basically that command reported in ERROR is just wrong, both syntactically and semantically.  It should be identical to what ktest executes.

**Comment by osa1** [comment]
This is related with https://github.com/kframework/k/pull/445/ . This should be a one line fix, we need to remove quoting first thing in the part between `[]`, implementation is here https://github.com/kframework/k/pull/445/files#diff-62188a71aec1c5c9c690d610f5b5113cR525

About sometimes redundant quoting of rest of the words, some discussions can be seen in #445 .

**Comment by osa1** [comment]
@grosu so basically we decided that quoting all words would be better ( see #445 ). But in the implementation first argument(which is actually the program name to run) is also quoted, which made the printed command invalid. If you remove quoting around first argument it should work fine.

Fixing this should take one line of change and I'll open a pull request unless someone else fixes this faster than me. (maybe @dwightguth push a fix faster without waiting for pull request etc.)

**Comment by grosu** [comment]
I don't think it is only one line, as I said before.  There is also the --directory option missing.  ktest is typically executed from a different folder from where the tested definition is (for example because it tests tens of definitions at the same time).  So the krun command should be self-contained and assumed to work from wherever it is called.  This has been done before by @daejunpark at my insistent requests and it worked fine.  I don't understand why it has been changed.  Please at least have Daejun revise changes to ktest.  Thanks!

**Comment by osa1** [comment]
Looks like that one(`--directory`) is broken with this change https://github.com/kframework/k/commit/f7b1e896c92794ff3a0aee47eaede2ff1e41a214#diff-a5cf3aaf006fff3d93af758c62573efdL308 . That patch was a bit problematic( #445 was also related with that ). I'll open a pull request about this two issues.

**Comment by osa1** [comment]
So I think we did two things wrong and broke a working program:
1. We failed at code reviews miserably. https://github.com/kframework/k/pull/295 the problematic patch was actually reviewed by both Daejun and me. (though reviewing it in details was very hard since the patch was huge)
2. We removed the code that we don't understand instead of asking the author for what it's doing.

**Comment by grosu** [comment]
Thanks, Omer!

Grigore

**Comment by dwightguth** [comment]
@grosu just a heads up, we pretty much need to keep the single quotes if you want it to work the way you are asking it to work. shell scripts on linux have a very wide range of different characters and behaviors which it would be ridiculous to try to emulate faithfully in all cases, and if we tried the end result would simply be that we missed some. So single quotes are used in order to make sure that the shell will process the command exactly as it was executed by ktest. Otherwise you have things like:

`krun -cPGM="$foo"`

which will quite unexpectedly treat foo as an environment variable.

**Comment by grosu** [comment]
I have absolutely no problem with the single quotes around the arguments of krun or kompile!  All I want is to make sure that I can copy the command enclosed by the ERROR[command] and paste it in the command line and execute it.  This is what I almost always do when kast gives me errors.

**Comment by osa1** [comment]
Fixed with #521.
